src/test/scala/chiselTests/Direction.scala

```diff
@@ -327,4 +327,73 @@ class DirectionSpec extends ChiselPropSpec with Matchers with Utils {
       }
     }
   }
+  property("Can now describe a Decoupled bundle using Flipped, not Input/Output in chisel3") {
+    class Decoupled extends Bundle {
+      val bits = UInt(3.W)
+      val valid = Bool()
+      val ready = Flipped(Bool())
+    }
+    class MyModule extends RawModule {
+      val incoming = IO(Flipped(new Decoupled))
+      val outgoing = IO(new Decoupled)
+
+      outgoing <> incoming
+    }
+
+    val emitted: String = ChiselStage.emitChirrtl(new MyModule)
+    val firrtl:  String = ChiselStage.convert(new MyModule).serialize
+
+    // Check that emitted directions are correct.
+    Seq(emitted, firrtl).foreach { o =>
+      {
+        // Chisel Emitter formats spacing a little differently than the
+        // FIRRTL Emitter :-(
```

Comment:
1. There isn't a Chisel emitter anymore so these emit exactly the same
2. Even if they did emit slightly differently, I don't think the purpose of these tests is to test the emitters so just using 1 is fine
3. Surprisingly, this test actually passes on both [3.6-SNAPSHOT](https://scastie.scala-lang.org/JXnauH3yQryVZzRMyNXkCA) and [3.5.3](https://scastie.scala-lang.org/wTroenCmSFmAO6R7xzSncQ) -- so that is weird...

Reply:
Conversation with @jackkoenig offline, realized this actually got introduced in 3.5.2 in the emitting firrtl bulk connect when possible update.